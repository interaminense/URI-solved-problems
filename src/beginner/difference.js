/**
 * Read four integer values named A, B, C and D. Calculate and print the difference of product A
 * and B by the product of C and D (A * B - C * D).
 *
 * INPUT
 *
 * The input file contains 4 integer values.
 * @param {number} a
 * @param {number} b
 * @param {number} c
 * @param {number} d
 *
 * OUTPUT
 *
 * Print DIFFERENCE with all the capital letters, according to the following example, with a blank
 * space before and after the equal signal.
 */

const tmpl = result => `DIFFERENCE = ${result}`;

const difference = (a, b, c, d) => tmpl((a * b - c * d));

export { difference, tmpl };
