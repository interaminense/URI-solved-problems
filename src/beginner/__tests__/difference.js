import { difference, tmpl } from '../difference';

describe('Difference', () => {
  let A = 0;
  let B = 0;
  let C = 0;
  let D = 0;
  const result = (a, b, c, d) => difference(a, b, c, d);

  it('The differente between 5, 6, 7 and 8 is -26', () => {
    A = 5;
    B = 6;
    C = 7;
    D = 8;

    expect(result(A, B, C, D)).toEqual(tmpl(-26));
    expect(result(A, B, C, D)).toMatchSnapshot();
  });

  it('The differente between 0, 0, 7 and 8 is -56', () => {
    A = 0;
    B = 0;
    C = 7;
    D = 8;

    expect(result(A, B, C, D)).toEqual(tmpl(-56));
    expect(result(A, B, C, D)).toMatchSnapshot();
  });

  it('The differente between 5, 6, -7 and 8 is 86', () => {
    A = 5;
    B = 6;
    C = -7;
    D = 8;

    expect(result(A, B, C, D)).toEqual(tmpl(86));
    expect(result(A, B, C, D)).toMatchSnapshot();
  });
});
