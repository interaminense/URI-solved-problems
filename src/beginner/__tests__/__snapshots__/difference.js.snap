// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Difference The differente between 0, 0, 7 and 8 is -56 1`] = `"DIFFERENCE = -56"`;

exports[`Difference The differente between 5, 6, -7 and 8 is 86 1`] = `"DIFFERENCE = 86"`;

exports[`Difference The differente between 5, 6, 7 and 8 is -26 1`] = `"DIFFERENCE = -26"`;
